v2/fb-xcom-api-validator/src/server/middlewares/auth-middleware.ts

```diff
@@ -0,0 +1,41 @@
+import { FastifyReply, FastifyRequest, HookHandlerDoneFunction } from 'fastify';
+import logger from '../../logging';
+import { HTTPMethod, verifySignature } from '../../auth-provider';
+import { InvalidSignatureError } from '../../signing';
+
+const log = logger('handler:auth');
+
+export function authMiddleware(request: FastifyRequest, reply: FastifyReply, done: HookHandlerDoneFunction) {
```

Comment:
We will need a separate middleware for each header component.

Reply:
renamed to `verifySignatureMiddleware`